EIPS/eip-remove_utf8_support.md

```diff
@@ -0,0 +1,140 @@
+## Preamble
+
+EIP: -
+
+Title: Remove UTF-8 support from contract strings
+
+Author: Andreas Olofsson (androlo1980@gmail.com)
+
+Type: Standard Track
+
+Category: ERC
+
+Status: Draft
+
+Created: 2017-10-16
+
+## Simple Summary
+
+This proposal is to remove UTF-8 support from contracts, and instead use a simpler and more generic string format.
+
+## Abstract
+
+Strings used in contracts are currently assumed to be UTF-8 encoded. There are no enforcement mechanisms, however, which means that a `string` is essentially the same as a `bytes`. An alternative to this is to define stings to be a more general form of `bytes`, which can be done by changing the the format of the header. The strings proposed here would allow user-defined alphabets with `O(1)` index access and length computation. Additionally, runtime validation of any given string would be done using a very simple compare function.
+
+## Motivation
+
+UTF-8 is not a good string format for contracts, for several reasons:
+
+1. It uses variable length characters, which means you have to index the strings before using them, or iterate over parts of the string when you want to get the length, or get a character by index.
+2. It has a non-trivial string validation scheme (http://www.unicode.org/versions/Unicode10.0.0/UnicodeStandard-10.0.pdf, Table 3-7, p 126, Well-Formed UTF-8 Byte Sequences).
+3. Strings that aren't written in plain English require more bytes, which means they incur a higher gas cost.
+
+These problems has lead to a marked lack of support for string processing and (runtime) validation in contract code.
```

Comment:
I don't think this is the reason for a lack of support for strings; lack of integrated language support plus a nascent ecosystem is the reason.

Reply:
yes that is what i meant. poorly worded. support as in "language X has built in support for working with utf-8 strings".